### PR TITLE
fix: make sure classes in lib correctly implement Electron interfaces

### PR DESCRIPTION
**Pull request by miniak**
We are already doing it in some places such as
https://github.com/electron/electron/blob/beb0cbc6d02815f5d4c6ed3dcdff931a5d8ba0cf/lib/browser/api/net-client-request.ts#L299

Let's do it consistently everywhere

Notes: none

### CODE REVIEW

File: lib/browser/api/auto-updater/auto-updater-win.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ class AutoUpdater extends EventEmitter {
   }
 
   getFeedURL () {
-    return this.updateURL;
+    return this.updateURL ?? '';
```
Comment (miniak) [inline lib/browser/api/auto-updater/auto-updater-win.ts:18]: the Mac implementation returns an empty string here if the URL is not specified. We should either do that here as well, or change the documentation to allow `null`

File: lib/browser/api/crash-reporter.ts [hunk 2]
```diff
@@ -45,7 +45,7 @@ class CrashReporter {
         return bts - ats;
       });
 
-    return (reports.length > 0) ? reports[0] : null;
+    return (reports.length > 0) ? reports[0] : null!;
```
Comment (miniak) [inline lib/browser/api/crash-reporter.ts:48]: The typings are being generated as `getLastCrashReport(): CrashReport;` with `null` missing

**Review by ckerr — APPROVED** [review]
The change LGTM.

Looks like the linux build has failing tests, but I can't tell if those are related to this change or not. Re-running them.

**Comment by miniak** [comment]
> The change LGTM.
> 
> Looks like the linux build has failing tests, but I can't tell if those are related to this change or not. Re-running them.

they are green now

**Pull request merged**